app/routes.py

```diff
@@ -1,2 +1,266 @@
-from flask import Blueprint
+from app import db 
+from flask import Blueprint, request, make_response, jsonify
+from app.models.task import Task 
+from app.models.goal import Goal
+from datetime import datetime
+import os 
+import requests
 
+
+
+task_list_bp = Blueprint("Task",__name__)
+
+@task_list_bp.route("/tasks", methods=["POST"])
+def create_a_task(): 
+    request_body = request.get_json()
+
+    if not ("title" in request_body
+            and "description" in request_body
+            and "completed_at" in request_body):
+    
+       
+         return make_response(jsonify({
+            "details": "Invalid data"
+        }), 400)
+        
+
+    new_task = Task.from_json(request_body)
+    # new_task = Task(title=request_body["title"],
+    #             description=request_body["description"],
+    #             completed_at=request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
```

Comment:
Just a suggestion, you could make saving a model to the DB an instance method in task, maybe:

```python
    def save(self):
        db.session.add(self)
        db.session.commit()
```

That way you:
1.  Keep the Database logic in the models and away from your routes
2. Dry up your routes code a little bit.